ground_truth_slam/plugin.cpp

```diff
@@ -26,29 +27,52 @@ class ground_truth_slam : public plugin {
 	}
 
 	void feed_ground_truth(const imu_cam_type *datum) {
-		ullong rounded_time = datum->dataset_time;
+		ullong rounded_time = floor(datum->dataset_time / 10000);
 		_m_sensor_data_it = _m_sensor_data.find(rounded_time);
 
 		if (_m_sensor_data_it == _m_sensor_data.end()) {
-			#ifndef NDEBUG
+#ifndef NDEBUG
 				std::cout << "True pose not found at timestamp: " << rounded_time << std::endl;
-			#endif
+#endif
 			return;
 		}
 
 		pose_type* true_pose = new pose_type{_m_sensor_data_it->second};
 		true_pose->sensor_time = datum->time;
-		// std::cout << "The pose was found at " << true_pose->position[0] << ", " << true_pose->position[1] << ", " << true_pose->position[2] << std::endl; 
-
 		_m_true_pose->put(true_pose);
+
+		// Ground truth position offset is the first ground truth position
+		if (first_time) {
+			first_time = false;
+			_m_ground_truth_offset->put(new Eigen::Vector3f{true_pose->position});
+		}
```

Comment:
```suggestion
		// Ground truth position offset is the first ground truth position
		if (first_time) {
			first_time = false;
			_m_ground_truth_offset->put(new Eigen::Vector3f{true_pose->position});
		}
		_m_true_pose->put(true_pose);
```

Reorder, because consumer assumes that `_m_true_pose` nonnull =>  `_m_ground_truth_offset` nonnull.